hack/e2e/run.sh

```diff
@@ -52,7 +52,7 @@ IMAGE_TAG=${IMAGE_TAG:-${TEST_ID}}
 # eksctl: mustn't include patch version (e.g. 1.19)
 K8S_VERSION=${K8S_VERSION:-1.20.8}
```

Comment:
This PR already implements several CI-related changes, maybe just bump this now?

Reply:
On second thought, this would require splitting the version for EKS/kops created clusters, probably better as its own PR